lib/cylc/scheduler_cli.py

```diff
@@ -437,15 +435,18 @@ def _rank_good_hosts(self, all_host_stats):
                 sys.stderr.write(base_msg + ', '.join(sort_asc_hosts) + '.\n')
             return sort_asc_hosts[0]
 
-    def appoint_host(self, mock_stats=False):
+    def appoint_host(self, mock_host_stats=None):
         """Appoint the most suitable host to (re-)run a suite on."""
+        if mock_host_stats is None and (
+                not self.hosts or self.hosts == ['localhost']):
```

Comment:
For simplicity we can drop the `self.hosts == ['localhost']` as that logic is covered by `self._trivial_choice`.